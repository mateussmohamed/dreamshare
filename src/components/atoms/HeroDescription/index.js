import React from 'react'
import styled from 'styled-components'
import { margin } from 'polished'

const HeroDescriptionStyled = styled.h4`
  ${margin('33px', '0')};
  ${props => props.theme.typography.robotoStyle};
  font-size: 24px;
  font-weight: 300;
  color: #f7f3f2;
`

const HeroDescription = props => (
  <HeroDescriptionStyled {...props}>{props.children}</HeroDescriptionStyled>
)

export default HeroDescription
