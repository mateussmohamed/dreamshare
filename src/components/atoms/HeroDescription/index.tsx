import React from 'react'

import styled from 'styled-components'

const HeroDescriptionStyled = styled.h4`
  ${props => props.theme.typography.robotoStyle};
  color: ${props => props.theme.colors.veryLightPink};
  font-size: 24px;
  font-weight: 300;

  @media only screen and (min-width: 320px) {
    font-size: 18px;
    margin: 20px 0;
  }

  @media only screen and (min-width: 768px) {
    margin: 30px 0;
    font-size: 24px;
  }
`

export const HeroDescription = props => (
  <HeroDescriptionStyled {...props}>{props.children}</HeroDescriptionStyled>
)
