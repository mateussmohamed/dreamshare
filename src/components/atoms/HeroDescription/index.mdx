---
name: HeroDescription
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import HeroDescription from './index'

# HeroDescription
<PropsTable of={HeroDescription} />

<Playground style={{ backgroundColor: 'tomato', padding: 40 }}>
  <HeroDescription>Find your holliday partner</HeroDescription>
</Playground>
