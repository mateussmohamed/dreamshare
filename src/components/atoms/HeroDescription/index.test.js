import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import HeroDescription from './index'

describe('[<HeroDescription />]', () => {
  const wrap = (props = {}) => shallow(<HeroDescription {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({ children: 'I am a description' })

    expect(wrapper).toMatchSnapshot()
  })
})
