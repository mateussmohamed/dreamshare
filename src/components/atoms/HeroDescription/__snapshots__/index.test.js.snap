// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<HeroDescription />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "HeroDescription__HeroDescriptionStyled-sc-1pz9ll8-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";color:",
          [Function],
          ";font-size:24px;font-weight:300;@media only screen and (min-width:320px){font-size:18px;margin:20px 0;}@media only screen and (min-width:768px){margin:30px 0;font-size:24px;}",
        ],
      },
      "displayName": "HeroDescription__HeroDescriptionStyled",
      "render": [Function],
      "styledComponentId": "HeroDescription__HeroDescriptionStyled-sc-1pz9ll8-0",
      "target": "h4",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  I am a description
</StyledComponent>
`;
